boa/src/syntax/ast/node/conditional.rs

```diff
@@ -0,0 +1,151 @@
+use super::Node;
+use gc::{Finalize, Trace};
+use std::fmt;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// The `if` statement executes a statement if a specified condition is [`truthy`][truthy]. If
+/// the condition is [`falsy`][falsy], another statement can be executed.
+///
+/// Multiple `if...else` statements can be nested to create an else if clause.
+///
+/// Note that there is no elseif (in one word) keyword in JavaScript.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#prod-IfStatement
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else
+/// [truthy]: https://developer.mozilla.org/en-US/docs/Glossary/truthy
+/// [falsy]: https://developer.mozilla.org/en-US/docs/Glossary/falsy
+/// [expression]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct If {
+    cond: Box<Node>,
+    body: Box<Node>,
+    else_node: Option<Box<Node>>,
+}
+
+impl If {
+    pub fn cond(&self) -> &Node {
+        &self.cond
+    }
+
+    pub fn body(&self) -> &Node {
+        &self.body
+    }
+
+    pub fn else_node(&self) -> &Option<Box<Node>> {
+        &self.else_node
+    }
```

Comment:
This too should return an `Option<&Node>`:
```suggestion
    pub fn else_node(&self) -> Option<&Node> {
        self.else_node.as_ref().map(Box::as_ref)
    }
```